src/bots/depositor.py

```diff
@@ -274,3 +275,57 @@ def _send_deposit_tx(
             self._flashbots_works = True
 
         return success
+
+    def _prepare_transaction(
+        self,
+        block_number: int,
+        block_hash: Hash32,
+        deposit_root: Hash32,
+        staking_module_id: int,
+        staking_module_nonce: int,
+        payload: bytes,
+        guardian_signs: tuple[tuple[str, str], ...]
+    ) -> ContractFunction:
+        """
+        It either follows a regular flow or builds a direct deposit transaction.
+
+        Conditions to build direct deposit transaction are:
+        1. Env variable is set
+        2. balance in the vault is not 0
+        3. The calls responded without errors
+        """
+        deposit_tx = self.w3.lido.deposit_security_module.deposit_buffered_ether(
+            block_number,
+            block_hash,
+            deposit_root,
+            staking_module_id,
+            staking_module_nonce,
+            payload,
+            guardian_signs,
+        )
+        if variables.is_mellow_strategy_set():
+            try:
+                vault_address = self.w3.lido.simple_dvt_staking_strategy.vault()
+                staking_module_contract: StakingModuleContract = self.w3.lido.simple_dvt_staking_strategy.staking_module_contract
+                if staking_module_contract.get_staking_module_id() != staking_module_id:
+                    logger.debug({'msg': 'While building mellow transaction module check failed.',
+                                  'contract_module': staking_module_contract.get_staking_module_id(),
+                                  'tx_module': staking_module_id})
+                    return deposit_tx
+                balance = staking_module_contract.weth_contract.balance_of(vault_address)
```

Comment:
Add new prometheus metric for Mellow Vault balance

Reply:
Done